restlicsharpclient/restliclient/transport/TransportResponse.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Newtonsoft.Json;
+
+using restlicsharpclient.restliclient.util;
+
+namespace restlicsharpclient.restliclient.transport
+{
+    public class TransportResponse
+    {
+        public Dictionary<string, object> data;
+        public Dictionary<string, string> headers;
+        public int? status;
+        // public Error error;
+        public bool fromNetwork;
+        // public errorDecoder = new ErrorResponseDecoder();
+
+        // Convert comma-separated wire header to app-expected header
+        public Dictionary<string, List<string>> responseHeaders
+        {
+            get
+            {
+                Dictionary<string, List<string>> appHeaders = new Dictionary<string, List<string>>();
+                foreach (KeyValuePair<string, string> pair in headers)
+                {
+                    appHeaders.Add(pair.Key, pair.Value.Split(RestConstants.kHeaderDelimiters).ToList());
+                }
+                return appHeaders;
+            }
+        }
+
+        public TransportResponse(HttpResponse response, bool fromNetwork = true)
+        {
+            this.fromNetwork = fromNetwork;
+
+            int? httpStatus = null;
+
+            // if response non-null, extract headers, status code, and data
+            if (response != null)
+            {
+                headers = response.headers;
+                httpStatus = response.status;
+                string dataString = System.Text.Encoding.UTF8.GetString(response.data);
+
+                data = JsonConvert.DeserializeObject<Dictionary<string, object>>(dataString, new JsonConverter[] { new DataMapDeserializationConverter() });
```

Comment:
Now I can see that you want to use this custom converter to handle nested array or record field in the JSON, but see my other comments in `DataMapDeserializationConverter`.